contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,807 @@
+     * requests from user level clients, for example in the kernel, IOService is a
+     * subclass of IORegistryEntry, which means any of the IORegistryEntryXXX
+     * functions in IOKitLib may be used with io_service_t's as well as
+     * io_registry_t's. There are functions available to introspect the class of the
+     * kernel object which any io_object_t et al. represents. IOKit objects returned
+     * by all functions should be released with {@link IOKit#IOObjectRelease}.
+     */
+    class IOObject extends PointerType {
+        public IOObject() {
+            super();
+        }
+
+        public IOObject(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link IOKit#IOObjectConformsTo} on this object.
+         *
+         * @param className
+         *            The name of the class.
+         * @return If the object handle is valid, and represents an object in the kernel
+         *         that dynamic casts to the class true is returned, otherwise false.
+         */
+        public boolean conformsTo(String className) {
+            return INSTANCE.IOObjectConformsTo(this, className);
+        }
+
+        /**
+         * Convenience method for {@link IOKit#IOObjectRelease} on this object.
+         *
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int release() {
+            return INSTANCE.IOObjectRelease(this);
+        }
+    }
+
+    /**
+     * An IOKit iterator handle.
+     */
+    class IOIterator extends IOObject {
+        public IOIterator() {
+            super();
+        }
+
+        public IOIterator(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link IOKit#IOIteratorNext} on this object.
+         *
+         * @return If the iterator handle is valid, the next element in the iteration is
+         *         returned, otherwise {@code null} is returned. The element should be
+         *         released by the caller when it is finished.
+         */
+        public IORegistryEntry next() {
+            return INSTANCE.IOIteratorNext(this);
+        }
+    }
+
+    /**
+     * The base class for all objects in the registry.
+     */
+    class IORegistryEntry extends IOObject {
+        public IORegistryEntry() {
+            super();
+        }
+
+        public IORegistryEntry(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetRegistryEntryID} to return
+         * an ID for this registry entry that is global to all tasks.
+         *
+         * @param id
+         *            The resulting ID.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getRegistryEntryID(LongByReference id) {
+            return INSTANCE.IORegistryEntryGetRegistryEntryID(this, id);
```

Comment:
Would it make to check the return of `IORegistryEntryGetRegistryEntryID` inside this method and return the resulting id from the method instead of the `k_return_t`? This assumes, that a fail is not expected and if it fails an Exception (see `com.sun.jna.platform.win32.Win32Exception` for inspiration).

Reply:
Good idea.